src/obs-ndi-source.cpp

```diff
@@ -1075,7 +1098,7 @@ void *ndi_source_create(obs_data_t *settings, obs_source_t *obs_source)
 		QString("OBS-NDI '%1'").arg(name).toUtf8();
 
 	// Allocate blank video frame
-	s->config.blank_frame = blank_video_frame();
+	// s->config.blank_frame = blank_video_frame();
```

Comment:
Intentionally not creating any blank_frame?
Does this work?

Reply:
Correct, the creation of the blank frame moved to ndi_source_update().

Reply:
This is the only line where `.blank_frame = anything` is called.  
There are no other calls to `blank_video_frame()`.  
So, nothing else is ever creating a blank frame.  
Meaning all uses of `s->config.blank_frame` are just sending nullptr.

Maybe this is intentional.  
Maybe this is accidental.  
Maybe you stumbled on a solution accidentally?

If not calling `blank_video_frame()` is intentional and fixes this, then maybe the whole `static obs_source_frame *blank_video_frame()` method should be removed?

Reply:
This is a happy little accident :D.

If we call "blank_video_frame()" = the bug exist. Setting config.blank_frame = nullptr (explicitly or not) fixes the problem.

I am looking though commit history to find the reason of that blank frame to see what it was used for. (fixing a bug maybe?)

Reply:
Digging deeper :

This blank_frame was added to fix the Audio only that was "keeping the last frame/not cleaning the frame". 
https://github.com/DistroAV/DistroAV/commit/7d1cf135df9610ae244c32282fdb1c9b838a5aca

This was part of v4.3.0 : "Source bugfix: video not cleared when switching to Audio Only mode"
https://github.com/DistroAV/DistroAV/releases/tag/4.3.0